waspc/src/Wasp/CompileOptions.hs

```diff
@@ -12,7 +12,8 @@ import Wasp.Message (SendMessage)
 --   It would be easier to pass around if it is part of Wasp data. But is it semantically correct?
 --   Maybe it is, even more than this!
 data CompileOptions = CompileOptions
-  { externalCodeDirPath :: !(Path' Abs (Dir SourceExternalCodeDir)),
+  { externalServerCodeDirPath :: !(Path' Abs (Dir SourceExternalCodeDir)),
+    externalClientCodeDirPath :: !(Path' Abs (Dir SourceExternalCodeDir)),
```

Comment:
@Martinsos We're dragging these two very similar (but different) properties (information about server code and client code) through many files in our codebase. Should I try somehow nesting them inside AppSpec and see if that helps with the duplication?

Reply:
I tried looking into it, but I haven't found many places -> from my search, they are defined here, then constructed in Compile.hs and Build.hs, and then used in Lib.hs . That goes for both of them. That is all that is happening with them, definition, two places where they are constructed, and then immediately consumed, and that is it.

Were you referring to smth else? I tried doing the same search for externalClientFiles, and externalServerFiles, but those are used at even less places.

Btw nesting them in AppSpec wouldn't make much sense -> AppSpec is defined after Analyzer, before the Generator, and used in Generator. But these two pieces of info never even get that far, they are consumed and not used anymore at the very beginning of the compiler itself.

Reply:
This, but also, `Lib.hs` then constructs two analogous fields inside AppSpec, which we then drag around too.

Reply:
I believe you are referring to externalClientCodeFiles and externalServerCodeFiles, as fields in AppSpec.
They are related to options from CompileOptions, yes, but they are different thing, they are actual files, while in CompileOptions it is just directory paths.

I don't understand where is the duplication, it seems like the right way to me, as it is done now, it is just the evolution of information. Can you explain it somehow differently to me? Where is the duplication, and how would you remove it?

Reply:
No no, it's alright. You had me at "evolution of information" :smile: